services/cd-service/pkg/repository/transformer.go

```diff
@@ -3717,9 +3723,17 @@ func (c *ReleaseTrain) Transform(
 	span.SetTag("environments", len(envNames))
 	if state.DBHandler.ShouldUseOtherTables() && isEnvGroup {
 		var wg sync.WaitGroup
```

Comment:
I think using an errgroup would be a lot simpler it has a max-concurrency feature build in.